typed/deduced_test.go

```diff
@@ -201,12 +201,9 @@ func TestMergeDeduced(t *testing.T) {
 			if err != nil {
 				t.Errorf("got validation errors: %v", err)
 			} else {
-				t.Logf("got:\v%v", got.AsValue())
-				gotUS := got.AsValue().ToUnstructured(true)
-				expectUS := out.AsValue().ToUnstructured(true)
-				if !reflect.DeepEqual(gotUS, expectUS) {
+				if !value.Equals(got.AsValue(), out.AsValue()) {
```

Comment:
Hopefully we can get `if !got.Equals(out) {`

Was the existing reflect.DeepEqual an oversight or deliberate?

Reply:
did you add Equals() to the interface?

Reply:
I did not. I think the deepequal was an oversight.